manticore/platforms/linux.py

```diff
@@ -1345,7 +1345,9 @@ def sys_lseek(self, fd, offset, whence):
             )
             return -e.err
 
-    def sys_llseek(self, fd, offset_high, offset_low, resultp, whence):
+    def sys_llseek(
+        self, fd: File, offset_high: int, offset_low: int, resultp: int, whence: int
```

Comment:
I think `fd` should just be an `int`, right? It's used as an index into an array here https://github.com/trailofbits/manticore/blob/bd28600ef35cdef06ab9522d4911a07350df3777/manticore/platforms/linux.py#L1253

And `self.files` is accessed using `int`, as shown when setting up the `stdio` `stdout` and `stderr` https://github.com/trailofbits/manticore/blob/bd28600ef35cdef06ab9522d4911a07350df3777/manticore/platforms/linux.py#L666-L671

Reply:
Yeah, you're right...

Reply:
Not sure why `mypy` isn't yelling at us though :(

Reply:
mypy by default doesn't do any checking within functions that don't have any type annotations.  Additionally, throughout the Linux platform code, there are very few type signatures, so most code in there gets an inferred type that allows anything.

I added several more type hints to things in the Linux platform code, for things that were obvious.  With those additions, mypy _did_ gripe about a few things, such as in `sys_write()`.  I fixed that.

Reply:
Awesome!! Thank you!!